[SYCL][Nativecpu] Native CPU backend build failure with --shared-libs

### Describe the bug

To reproduce:

Here is the configure line:
```
CUDA_LIB_PATH=${CUDA_HOME} CC=`which gcc` CXX=`which g++` python3 $DPCPP_HOME/llvm/buildbot/configure.py --shared-libs --cmake-gen "Unix Makefiles" -DCUDAToolkit_ROOT=${CUDA_HOME} -DCMAKE_INSTALL_PREFIX=/soft/compilers/oneapi/upstream/install-${cuda_version}_${NOW} -DCMAKE_C_COMPILER=`which gcc` -DCMAKE_CXX_COMPILER=`which g++` --native_cpu --cuda -t Release --llvm-external-projects openmp
```

(sample) Errors:
```
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: sub_group_analysis.cpp:(.text._ZN8compiler5utils18GlobalSubgroupInfoC2ERN4llvm6ModuleERNS0_11BuiltinInfoE+0x999): undefined reference to `llvm::SmallPtrSetImplBase::insert_imp_big(void const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: sub_group_analysis.cpp:(.text._ZN8compiler5utils18GlobalSubgroupInfoC2ERN4llvm6ModuleERNS0_11BuiltinInfoE+0x9dc): undefined reference to `llvm::SmallPtrSetImplBase::insert_imp_big(void const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: sub_group_analysis.cpp:(.text._ZN8compiler5utils18GlobalSubgroupInfoC2ERN4llvm6ModuleERNS0_11BuiltinInfoE+0xdce): undefined reference to `llvm::deallocate_buffer(void*, unsigned long, unsigned long)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: sub_group_analysis.cpp:(.text._ZN8compiler5utils18GlobalSubgroupInfoC2ERN4llvm6ModuleERNS0_11BuiltinInfoE+0x12a2): undefined reference to `llvm::SmallVectorBase<unsigned int>::grow_pod(void*, unsigned long, unsigned long)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: sub_group_analysis.cpp:(.text._ZN8compiler5utils18GlobalSubgroupInfoC2ERN4llvm6ModuleERNS0_11BuiltinInfoE+0x12c5): undefined reference to `llvm::SmallVectorBase<unsigned int>::grow_pod(void*, unsigned long, unsigned long)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: sub_group_analysis.cpp:(.text._ZN8compiler5utils18GlobalSubgroupInfoC2ERN4llvm6ModuleERNS0_11BuiltinInfoE+0x1420): undefined reference to `llvm::SmallVectorBase<unsigned int>::grow_pod(void*, unsigned long, unsigned long)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: sub_group_analysis.cpp:(.text._ZN8compiler5utils18GlobalSubgroupInfoC2ERN4llvm6ModuleERNS0_11BuiltinInfoE+0x1446): undefined reference to `llvm::SmallVectorBase<unsigned int>::grow_pod(void*, unsigned long, unsigned long)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: sub_group_analysis.cpp:(.text._ZN8compiler5utils18GlobalSubgroupInfoC2ERN4llvm6ModuleERNS0_11BuiltinInfoE+0x1469): undefined reference to `llvm::SmallVectorBase<unsigned int>::grow_pod(void*, unsigned long, unsigned long)'
```

Attached is the full log of the error:

[sycl_native_cpu.txt](https://github.com/user-attachments/files/22285253/sycl_native_cpu.txt)

### To reproduce

1. Include a code snippet that is as short as possible
2. Specify the command which should be used to compile the program
3. Specify the command which should be used to launch the program
4. Indicate what is wrong and what was expected


### Environment

- OS: [e.g Windows/Linux]
- Target device and vendor: [e.g. Intel GPU]
- DPC++ version: [e.g. commit hash or output of `clang++ --version`]
- Dependencies version: [e.g. the output of `sycl-ls --verbose`]


### Additional context

_No response_

--- a/llvm/lib/SYCLNativeCPUUtils/CMakeLists.txt
+++ b/llvm/lib/SYCLNativeCPUUtils/CMakeLists.txt
@@ -21,15 +21,17 @@ add_llvm_component_library(LLVMSYCLNativeCPUUtils
   ${LLVM_MAIN_INCLUDE_DIR}/llvm/SYCLLowerIR
   
   LINK_COMPONENTS
+  AggressiveInstCombine
   Analysis
   Core
-  Support
+  IPO
   Passes
+  ScalarOpts
+  Support
   SYCLLowerIR
   Target
   TargetParser
   TransformUtils
-  ipo
   ${OCK_LIBS}
 )
 

--- a/llvm/lib/SYCLNativeCPUUtils/compiler_passes/compiler_pipeline/CMakeLists.txt
+++ b/llvm/lib/SYCLNativeCPUUtils/compiler_passes/compiler_pipeline/CMakeLists.txt
@@ -21,8 +21,11 @@ add_llvm_component_library(LLVMNativeCPUPipeline
   ${CMAKE_CURRENT_SOURCE_DIR}/source/work_item_loops_pass.cpp
 
   LINK_COMPONENTS
-  Passes
+  Analysis
   Core
+  Passes
+  Support
+  TransformUtils
   )
 
 # TODO: Move to under LLVM include and work out why ADDITIONAL_HEADER_DIRS

--- a/llvm/lib/SYCLNativeCPUUtils/compiler_passes/vecz/CMakeLists.txt
+++ b/llvm/lib/SYCLNativeCPUUtils/compiler_passes/vecz/CMakeLists.txt
@@ -106,16 +106,18 @@ endif()
 add_llvm_component_library(LLVMNativeCPUVecz
   ${COMMON_SRCS}
   LINK_COMPONENTS
+  AggressiveInstCombine
+  Analysis
+  Core
+  InstCombine
+  IPO
   NativeCPUPipeline
-  support
-  core
-  analysis
-  instcombine
-  aggressiveinstcombine
-  transformutils
-  scalaropts
-  ipo
-  passes
+  Passes
+  ScalarOpts
+  Support
+  Target
+  TargetParser
+  TransformUtils
   )
 
 target_include_directories(LLVMNativeCPUVecz

